### DISCUSSION TITLE
UserWarning: The setup method 'route' can no longer be called on the blueprint 'auth'

### DISCUSSION THREAD

**Discussion by Balinty** [pallets/flask#5015] [Q&A] [answered]
I started to get this error from all of my blueprints:

> UserWarning: The setup method 'route' can no longer be called on the blueprint 'auth'. It has already been registered at least once, any changes will not be applied consistently. Make sure all imports, decorators, functions, etc. needed to set up the blueprint are done before registering it. This warning will become an exception in Flask 2.3.

Is there a way to fix this?

Code from one of my blueprints:

```
from flask import Blueprint, current_app, render_template, request, url_for, redirect, session
auth = Blueprint('auth', __name__)

@auth.route("/xy")
def xy():
...
```
I configured all of my blueprints like this:

```
from flask import Flask

def create_app():
    app = Flask(__name__, template_folder='www', static_folder='static')

    from auth import auth
    from app import weather
    from api import api
    from search import searchapp

    app.register_blueprint(weather)

    app.register_blueprint(auth)

    app.register_blueprint(api)

    app.register_blueprint(searchapp)

    return app

```

**Comment by davidism** [comment, marked as answer]
Please include a minimal reproducible example, and include the traceback with the warning. The code you've shown does not demonstrate the error.

In general, you fix this in exactly the way that the message describes. Do not call any methods that set up the blueprint, such as registering routes, error handlers, before request functions, etc. after calling `register_blueprint` on it.

**Reply by Balinty** [reply]
Then what should i use instead of @auth.route()?

**Reply by davidism** [reply]
That's not what the error says. It says to do all setup before registering the blueprint, not to stop using the route decorator.

**Comment by githubkind** [comment]
![1684122308405](https://github.com/pallets/flask/assets/85106789/7700907c-9474-4979-89e2-4a2e36f78285)
after I comment out this line，it works

**Reply by ByteDevMaven** [reply]
How did you import it then?
